www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -84,33 +84,77 @@ function shouldNotBeEqTo0($value)
 
 function multipleServiceCategorieInDB($sc = array(), $nbrDup = array())
 {
-        global $pearDB, $centreon;
-                
-        $scAcl = array();
+    global $pearDB, $centreon;
+
+    $scAcl = array();
     foreach ($sc as $key => $value) {
-        $DBRESULT = $pearDB->query("SELECT * FROM `service_categories` WHERE `sc_id` = '".$key."' LIMIT 1");
-        $row = $DBRESULT->fetchRow();
+        $scId = filter_var($key, FILTER_VALIDATE_INT);
+        $statement = $pearDB->prepare("SELECT * FROM `service_categories` WHERE `sc_id` = ? LIMIT 1");
+        $result = $pearDB->execute($statement, array($scId));
+        $row = $result->fetchRow();
         $row["sc_id"] = '';
         for ($i = 1; $i <= $nbrDup[$key]; $i++) {
             $val = null;
+            $params = array();
             foreach ($row as $key2 => $value2) {
-                $key2 == "sc_name" ? ($sc_name = $value2 = $value2."_".$i) : null;
-                $key2 == "sc_description" ? ($sc_alias = $value2 = $value2) : null;
-                $val ? $val .= ($value2!=null?(", '".$value2."'"):", NULL") : $val .= ($value2 != null?("'".$value2."'"):"NULL");
+                switch ($key2) {
+                    case 'sc_name':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $sc_name = $value2 = $value2 . "_" . $i;
+                        $params[] = $value2;
+                        break;
+                    case 'sc_description':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $params[] = $value2;
+                        break;
+                    case 'level':
```

Comment:
You can combine this 2 tests:

                    case 'level':
                    case 'icon_id':
                        $value2 = filter_var($value2, FILTER_VALIDATE_INT);
                        $value2
                            ? $params[] = $value2
                            : $params[] = null;
                        break;

Reply:
done